### ISSUE TITLE
bug(semantic): params in `export default (function() {})` flagged as `SymbolFlags::Export`

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#4475]
> Found in #4445 

Parameters in parenthesis-wrapped function expressions in `ExportDefaultDeclaration`s are being incorrectly flagged as exports.
```ts
// correct: `a` is not flagged as exported
export default function (a) { }
// incorrect: `a` is flagged as exported
export default (function(a) {});
```

![image](https://github.com/user-attachments/assets/c907f2ae-137c-4e5d-a087-d9234d2a69ec)
